accounting/filters/ChtcScheddCpuOspoolFilter.py

```diff
@@ -0,0 +1,857 @@
+            self.schedd_collector_host_map[schedd] = set()
+            self.schedd_collector_host_map_checked.add(schedd)
+            new_hosts = False
+
+            collectors_queried = set()
+            for collector_host in self.collector_hosts:
+                if collector_host in {"flock.opensciencegrid.org"}:
+                    continue
+                collector = htcondor.Collector(collector_host)
+                try:
+                    ads = collector.query(
+                        htcondor.AdTypes.Schedd,
+                        constraint=f'''Machine == "{schedd.split('@')[-1]}"''',
+                        projection=["CollectorHost"],
+                    )
+                except htcondor.HTCondorIOError:
+                    continue
+                collectors_queried.add(collector_host)
+                ads = list(ads)
+                if len(ads) == 0:
+                    continue
+                if len(ads) > 1:
+                    self.logger.warning(f'Got multiple Schedd ClassAds for Machine == "{schedd}"')
+
+                # Cache the CollectorHost in the map
+                if "CollectorHost" in ads[0]:
+                    schedd_collector_hosts = set()
+                    for schedd_collector_host in re.split(r'[, ]+', ads[0]["CollectorHost"]):
+                        schedd_collector_host = schedd_collector_host.strip().split(":")[0]
+                        if schedd_collector_host:
+                            schedd_collector_hosts.add(schedd_collector_host)
+                            new_hosts = True
+                    if schedd_collector_hosts:
+                        self.schedd_collector_host_map[schedd] = schedd_collector_hosts
+                        break
+            else:
+                self.logger.warning(f"Did not find Machine == {schedd} in collectors {', '.join(collectors_queried)}")
+
+            # Update the pickle
+            if new_hosts and len(schedd_collector_hosts) > 0:
+                self.logger.debug(f"Updating collector host pickle for {schedd} with {schedd_collector_hosts}")
+                old_schedd_collector_host_map = {}
+                if self.schedd_collector_host_map_pickle.exists():
+                    try:
+                        old_schedd_collector_host_map = pickle.load(open(self.schedd_collector_host_map_pickle, "rb"))
+                        old_schedd_collector_host_map[schedd] = schedd_collector_hosts
+                    except IOError:
+                        pass
+                old_schedd_collector_host_map[schedd] = schedd_collector_hosts
+                with open(self.schedd_collector_host_map_pickle, "wb") as f:
+                    pickle.dump(old_schedd_collector_host_map, f)
+
+        return self.schedd_collector_host_map[schedd]
+
+    def is_ospool_job(self, ad):
+        remote_pool = set()
+        if "LastRemotePool" in ad and ad["LastRemotePool"]:
+            remote_pool.add(ad["LastRemotePool"])
+        else:
```

Comment:
This `else` block isn't technically necessary here since _currently_ none of the `chtc.wisc.edu` APs will have the OSPool central manager as their `CollectorHost`, but it's probably good to leave it in for future-proofing in case that ever changes.